integration-tests/validation/src/main/java/com/android/designcompose/testapp/validation/MainActivity.kt

```diff
@@ -437,13 +442,27 @@ fun VConstraintsTest() {
 // TEST Interactions
 @DesignDoc(id = "8Zg9viyjYTnyN29pbkR1CE")
 interface InteractionTest {
-    @DesignComponent(node = "Start Here") fun MainFrame()
+    @DesignComponent(node = "Start Here")
+    fun MainFrame(
+        @Design(node = "#KeyButtonB") onTapKeyButtonB: TapCallback,
+        @Design(node = "#KeyButtonC") onTapKeyButtonC: TapCallback,
+    )
+
+    // Inject a ctrl-shift-B key when the 'clickedB()' function is called
+    @DesignKeyAction(key = 'B', metaKeys = [DesignMetaKey.MetaShift, DesignMetaKey.MetaCtrl])
+    fun clickedB()
+    // Inject a meta-C key when the 'clickedC()' function is called
+    @DesignKeyAction(key = 'C', metaKeys = [DesignMetaKey.MetaMeta]) fun clickedC()
 }
 
 @Preview
 @Composable
 fun InteractionTest() {
-    InteractionTestDoc.MainFrame()
+    val rootView = LocalView.current
+    InteractionTestDoc.MainFrame(
+        onTapKeyButtonB = { InteractionTestDoc.clickedB() },
```

Comment:
Yep, this is about as nice as we can make this without being able to register custom events for the Figma UI to show...

If we have multiple instances of a component, then I think they're all sharing the same `InteractionState` instance, so that means that the generated methods can be static, but it also means that we can't have two independent instances of a component. This isn't a new issue, and it hasn't hurt us much before (does it break pressed state for list items?).

If we want to fix this global state issue, then we'll need to pass in `InteractionState` (or some "context"/"identity" object that just contains a unique key) as an argument to the component and as an argument to the generated key action functions; maybe make a follow-up ticket on this?